workers/generic-worker/mounts.go

```diff
@@ -855,12 +858,13 @@ func (ic *IndexedContent) String() string {
 	return "namespace " + ic.Namespace + " artifact " + ic.Artifact
 }
 
-func (ac *ArtifactContent) UniqueKey() string {
-	return "artifact:" + ac.TaskID + ":" + ac.Artifact
+func (ac *ArtifactContent) UniqueKey(taskMount *TaskMount) (string, error) {
+	return "artifact:" + ac.TaskID + ":" + ac.Artifact, nil
 }
 
-func (ic *IndexedContent) UniqueKey() string {
-	return "indexed:" + ic.Namespace + ":" + ic.Artifact
+func (ic *IndexedContent) UniqueKey(taskMount *TaskMount) (string, error) {
+	itr, err := taskMount.index.FindTask(ic.Namespace)
+	return "indexed:" + ic.Namespace + ":" + ic.Artifact + ":" + itr.TaskID, err
```

Comment:
```suggestion
	return "artifact:" + itr.TaskID + ":" + ic.Artifact, err
```

This way, we won't have two copies of the same artifact in the disk cache, if they represent the same content, but one is an indexed artifact, and the other explicitly refers to the task id. Also if there are two separate indexes (e.g. a nightly index, and a latest index) that happen to point to exactly the same taskId/artifact combination, this also won't be duplicated in the disk cache.